Some UI tweaks / fix EMC2101 fan PWM setting / add ASIC temperature set point option

Rob0xFF opened the pull request:
- AxeOS: calculate value to be written to EMC2101_REG_FAN_SETTING register by considering EMC2101_PWM_FREQ according to eq. 6, see EMC2101 datasheet appendix A

for example, a fan setting of 50% yielded approx. 70% of fan RPM before, now it correctly yields 50% of fan's max RPM

- AxeOS: EMC2101 register value EMC2101_INTERNAL_TEMP is 8bit signed, changed return value accordingly, now supporting temperatures below 0°C

- AxeOS: tweaked min_fan_speed in power management

- UI: moved units from inside the p-knop to the outside to ensure they do not overlap the knop graphic

- UI/AxeOS: added option to define a setpoint ASIC temperature to maintain

- UI: added groups to setting UI for better overview / readability

- AxeOS: using EMC2101_FAN_RPM_NUMERATOR instead of explicit integer

- Github: changed instruction to build firmware

- fix flip screen not working

- convert all c array images to pure true color without alpha channel to save 10% of flash space

- fix unit of measure for efficiency on mining screen

Rob0xFF committed:
Minor UI Tweaks

Moved units from inside the p-knop to the outside to ensure they do not overlap the knop graphic

Rob0xFF committed:
fixed EMC2101 fan PWM setting

- calculate value to be written to EMC2101_REG_FAN_SETTING register by considering EMC2101_PWM_FREQ according to eq. 6, see datasheet appendix A

for example, a fan setting of 50% yielded approx. 70% before, now it correctly yields 50% of fan's max RPM

- EMC2101 register value EMC2101_INTERNAL_TEMP is 8bit signed, changed return value accordingly, now supporting temperatures below 0°C

- tweaked min_fan_speed in power management

Rob0xFF committed:
Update readme.md

add instructions on how to build firmware

Rob0xFF committed:
Added ASIC temperature setpoint control option

- UI/AxeOS: added option to define a setpoint ASIC temperature to maintain
- UI: added groups to setting UI for better overview / readability
- AxeOS: using EMC2101_FAN_RPM_NUMERATOR instead of explicit integer

Rob0xFF committed:
UI tweaks

- dashboard page: layout fixes for small displays

Rob0xFF committed:
fix flip screen not working / minor UI fixes

- flip screen now working for NerdAxe T-Display (button did not do anything on NerdAxe)
- minor UI fixes to give a more unified look

Rob0xFF committed:
UI

Rob0xFF committed:
optimize image C arrays / fix efficiency unit on mining screen

- converted all c arrays to true color without alpha channel, saving about 220 kbyte of flash
- fixed efficiency unit to J/kH on mining screen

skot committed:
clear ASIC RX serial buffer on framing errors (#279)

* added rx serial buffer flush and serial debugging for BM1368

* added serial buffer flush to BM1366 also

* turned off ASIC serial debugging. let's see how this does on free heap

(cherry picked from commit 2e03a8f0e96fdc8c6064b1643ecc11743698e0c2)
(cherry picked from commit cf99b50faffb68239356f51e38f4fb4a4ce7651d)

Rob0xFF committed:
restructured oled and ttgo code a liiiitle bit

(cherry picked from commit 3d17f039c481b0b7600fea39612b222f813f3d77)

# Conflicts:
#	main/system.c
(cherry picked from commit 3ea05b9ccf1fae20101559f291de666c749bd61e)

Rob0xFF committed:
fix clock sync on every nonce time to sync every hour as intended

(cherry picked from commit 545d91663ec3e8725bfadcac68cc6149790d4a93)

Rob0xFF committed:
fix fan too low if setpoint control and temp low

(cherry picked from commit ec984824ad601192b645644857f62cfd41dd3c49)

Rob0xFF committed:
Revert "fix clock sync on every nonce time to sync every hour as intended"

This reverts commit 84d3bd020bffa3c012fed117bfe34cf6b8002839.

Rob0xFF committed:
new fix for clock sync

Rob0xFF committed:
omit explicit search for init value of last clock sync

shufps commented on main/displays/displayDriver.c:
```diff
@@ -396,6 +396,8 @@ lv_obj_t * initTDisplayS3(void){
     disp_drv.flush_cb = example_lvgl_flush_cb;
     disp_drv.draw_buf = &disp_buf;
     disp_drv.user_data = panel_handle;
+    disp_drv.sw_rotate = 1;
```
The display supports hardware rotation.

try this instead:
```
    if (!nvs_config_get_u16(NVS_CONFIG_FLIP_SCREEN, 0)) {
        esp_lcd_panel_mirror(panel_handle, true, false);
    } else {
        esp_lcd_panel_mirror(panel_handle, false, true);
    }
    ```

Rob0xFF replied:
Thanks, I have to try your code a.s.a.p. I tried to do it the hardware way with no success, so I chose software rotation.

BitMaker-hub commented:
Amazing work, guys, thank you. I'll test this today

Rob0xFF commented:
I am still working on a means to automatically switch on and off the screen based on time of day. Therefore, time zone must be set by the user and the user must choose the on and off time. At the moment, both is hard coded in the test branch of my fork. Works for me at least. Additionally, I implemented a screen notification, if the hash difficulty > network difficulty, i.e. a block is found. It is also shown in the web UI. I added some more stats to the Logs section of the UI, like NerdDisplay status and reason of last reset. All of this is in the test branch of my fork,

Please let me know, if you guys would be interested. I could try to develop further.

Rob0xFF commented:
I also set up my personal node using umbrella and am mining to public pool on the node to test performance. I have seen, that using the default string in stratum_api.c from the original Nerdaxe repository

`sprintf(subscribe_msg, "{\"id\": %d, \"method\": \"mining.subscribe\", \"params\": [\"bitaxe/%s (%s)\"]}\n", send_uid++, model, version);`

yields that my Nerdaxe is identified as "Bitaxe" on public pool. So, as I told on X, I think all those NerdAxes shown on public-pool.io might not be the ones of this project as they all seem to identify as Bitaxe. If I am right, we should introduce a unique identifier when subscribing to pool via stratum in order to distinguish Nerdaxes from this project and other ones. There is an other project, which also calls itself "Nerdaxe", as you know.

Rob0xFF commented:
Also a to do is to let the user set the max fan speed of his device, to improve UI graphic, i.e. full use of the round bar should match rpm which equals 100% fan speed. I have a noctua with 5000rpm, so I hardcoded it for the moment.

Keep up the great work guys.

BitMaker-hub commented:
Allright, I've just seen I made a mistake when I rebased last firmware to last bitaxe changes. 
I mainly rewritted the mining suscribe method from nerdaxe to bitaxe.
I've just changed back again.

I will try to check the changes you did this week and come back with a new screen and a new update.

Rob0xFF commented:
Okay, thanks for clearing things up regarding the identifier. Looking forward to the next release. Really love the Nerdaxe, so keep up the good work.BTW if you plan to upgrade to other ASICS, please consider aligning the visible display area of the Lilygo to the fan to give a more clear look. I tried to work around this with my case [here](https://www.printables.com/de/model/947709-nerdaxe-case-with-stand)
